Issue1253 read matrix functions

jhmigueles opened the pull request:
<!-- Describe your PR here -->
Fixes #1253 => Implements functionality to read and process Parmay Matrix sensors data.


<!-- Please, make sure the following items are checked -->
### Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [x] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Clean code has been attempted, e.g. intuitive object names and no code redundancy.
- [x] Documentation updated:
  - [x] Function documentation
  - [x] Chapter vignettes for GitHub IO
  - [x] Vignettes for CRAN
- [x] Corresponding issue tagged in PR message. If no issue exist, please create an issue and tag it.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.
- [ ] GGIR parameters were added/removed. If yes, please also complete checklist below.

**If NEW GGIR parameter(s) were added then these NEW parameter(s) are:**
- [ ] documented in `man/GGIR.Rd`
- [ ] included with a default in `R/load_params.R`
- [ ] included with value class check in `R/check_params.R`
- [ ] included in table of `vignettes/GGIRParameters.Rmd` with references to the GGIR parts the parameter is used in.
- [ ] mentioned in NEWS.Rd as NEW parameter

**If GGIR parameter(s) were deprecated these parameter(s) are:**
- [ ] documented as deprecated in `man/GGIR.Rd`
- [ ] removed from `R/load_params.R`
- [ ] removed from `R/check_params.R`
- [ ] removed from table in `vignettes/GGIRParameters.Rmd`
- [ ] mentioned as deprecated parameter in NEWS.Rd
- [ ] added to the list in `R/extract_params.R` with deprecated parameters such that these do not produce warnings when found in old config.csv files.

vincentvanhees reviewed (changes requested):
Hi Jairo, my specific comments are minor and hopefully self-explanatory. Additionally, please find below my general comments which relate to parts of the code you did not change, mostly related to documentation:

1. Please expand unit test [test_greadaccfile.R](https://github.com/wadpac/GGIR/blob/master/tests/testthat/test_greadaccfile.R) to cover Parmay Matrix. As you will see, we do similar checks there for Axivity. The focus of this unit-test is not to check whether we can read the data, because GGIRread already does that, but whether g.readaccfile and g.inspectfile do they what we expect. You can use the GGIRread test files, so no need to add extra test files to GGIR. Make sure that the new `inspect_binFile_brand.R` is also covered.
2. Add Matrix to [chapter 2](https://wadpac.github.io/GGIR/articles/chapter2_Pipeline.html) text and table in 
3. Expand documentation for interpolationtype to also include Matrix.
4. Check that sections in [chapter 3](https://wadpac.github.io/GGIR/articles/chapter3_QualityAssessment.html) relevant to Martrix are updated/expanded.
5. Is LUX data captured and shared with GGIR? If yes, please also update corresponding section in [chapter 13](https://wadpac.github.io/GGIR/articles/chapter13_CircadianRhythm.html?q=Axivity#mxlx).
6. Please update [CRAN vignette monitor brand list](https://github.com/wadpac/GGIR/blob/master/vignettes/GGIR.Rmd) with Parmay Matrix.
7. When I write `closes #` followed by issue number I do it to close an issue without considering it fixed, and when I use `fixes #` followed by issue number to indicate that I fixed the issue. In both cases the issue will be closed. There is no convention about this, but if you like my reasoning feel free to copy (no pressure!).

jhmigueles commented on DESCRIPTION:
```diff
@@ -37,6 +37,6 @@ URL: https://github.com/wadpac/GGIR/, https://groups.google.com/forum/#!forum/Rp
 License: Apache License (== 2.0) | file LICENSE
 Suggests: testthat, covr, knitr, rmarkdown, actilifecounts, readxl
-Imports: data.table, foreach, doParallel, signal, zoo, unisensR, ineq, methods, psych, irr, lubridate, GGIRread, ActCR, read.gt3x
+Imports: data.table, foreach, doParallel, signal, zoo, unisensR, ineq, methods, psych, irr, lubridate, GGIRread (>= 1.0.4), ActCR, read.gt3x
```
I considered this but wasn't sure how to do it. The problem is that to know that the input files are matrix files we need to try to read the file, we cannot know this from user input parameters. So, we cannot foresee this in the GGIR function, we would need to trigger the error in g.inspectfile. I have now implemented it, let me kow if you prefer this otherwise.

vincentvanhees replied:
I have now addressed this in [8b71ccc](https://github.com/wadpac/GGIR/pull/1286/commits/8b71ccc53bba99997ece8291498a3c07fc6d376a)

jhmigueles commented on NEWS.md:
```diff
@@ -1,3 +1,7 @@
+# CHANGES IN GGIR VERSION 3.2-?
+
+- Part 1: Implemented functionality to read and process Parmay Matrix sensors data (BIN files) #1253
```
Updated NEWS and created the issue.

jhmigueles commented:
- [ ] Please expand unit test [test_greadaccfile.R](https://github.com/wadpac/GGIR/blob/master/tests/testthat/test_greadaccfile.R) to cover Parmay Matrix. As you will see, we do similar checks there for Axivity. The focus of this unit-test is not to check whether we can read the data, because GGIRread already does that, but whether g.readaccfile and g.inspectfile do they what we expect. You can use the GGIRread test files, so no need to add extra test files to GGIR. Make sure that the new inspect_binFile_brand.R is also covered.

- [ ] Add Matrix to [chapter 2](https://wadpac.github.io/GGIR/articles/chapter2_Pipeline.html) text and table in

- [ ] Expand documentation for interpolationtype to also include Matrix.

- [ ] Check that sections in [chapter 3](https://wadpac.github.io/GGIR/articles/chapter3_QualityAssessment.html) relevant to Martrix are updated/expanded.

- [ ] Is LUX data captured and shared with GGIR? If yes, please also update corresponding section in [chapter 13](https://wadpac.github.io/GGIR/articles/chapter13_CircadianRhythm.html?q=Axivity#mxlx).

Matrix devices do not have lux sensors.

- [ ] Please update [CRAN vignette monitor brand list](https://github.com/wadpac/GGIR/blob/master/vignettes/GGIR.Rmd) with Parmay Matrix.

- [ ] When I write closes # followed by issue number I do it to close an issue without considering it fixed, and when I use fixes # followed by issue number to indicate that I fixed the issue. In both cases the issue will be closed. There is no convention about this, but if you like my reasoning feel free to copy (no pressure!).

I think it is a good strategy, I modified the comment in the PR =)
